egui_demo_lib/src/apps/demo/zoom_rotate.rs

```diff
@@ -0,0 +1,117 @@
+use egui::{
+    emath::{RectTransform, Rot2},
+    vec2, Color32, Frame, Pos2, Rect, Sense, Stroke,
+};
+
+pub struct ZoomRotate {
+    time_of_last_update: Option<f64>,
+    rotation: f32,
+    zoom: f32,
+}
+
+impl Default for ZoomRotate {
+    fn default() -> Self {
+        Self {
+            time_of_last_update: None,
+            rotation: 0.,
+            zoom: 1.,
+        }
+    }
+}
+
+impl super::Demo for ZoomRotate {
+    fn name(&self) -> &'static str {
+        "👌 Multi Touch"
+    }
+
+    fn show(&mut self, ctx: &egui::CtxRef, open: &mut bool) {
+        egui::Window::new(self.name())
+            .open(open)
+            .default_size(vec2(512.0, 512.0))
+            .resizable(true)
+            .show(ctx, |ui| {
+                use super::View;
+                self.ui(ui);
+            });
+    }
+}
+
+impl super::View for ZoomRotate {
+    fn ui(&mut self, ui: &mut egui::Ui) {
+        ui.vertical_centered(|ui| {
+            ui.add(crate::__egui_github_link_file!());
+        });
+        ui.colored_label(
+            Color32::RED,
+            "This only works on devices which send native touch events (mostly mobiles).",
+        );
+        ui.separator();
+        ui.label("Try touch gestures Pinch/Stretch, Rotation, and Pressure with 2+ fingers.");
+        Frame::dark_canvas(ui.style()).show(ui, |ui| {
+            // Note that we use `Sense::drag()` although we do not use any pointer events.  With
+            // the current implementation, the fact that a touch event of two or more fingers is
+            // recognized, does not mean that the pointer events are suppressed, which are always
+            // generated for the first finger.  Therefore, if we do not explicitly consume pointer
+            // events, the window will move around, not only when dragged with a single finger, but
+            // also when a two-finger touch is active.  I guess this problem can only be cleanly
+            // solved when the synthetic pointer events are created by egui, and not by the
+            // backend.
```

Comment:
👍  I think it makes sense to consider multitouch manipulation as a form of dragging.